crates/phactory/src/lib.rs

```diff
@@ -203,10 +205,12 @@ impl<Platform: pal::Platform> Phactory<Platform> {
         }
 
         self.args = args;
+        self.attestation_provider = self.args.attestation_provider.clone();
     }
 
     pub fn set_args(&mut self, args: InitArgs) {
         self.args = args;
+        self.attestation_provider = self.args.attestation_provider.clone();
```

Comment:
The `set_args` is called immediately after restored from checkpoint. So the `attestation_provider` should keep the value from checkpoint. Infact, any parameters that could not be changed over a restoring should not be put in the command line args. Better put it in the `InitRuntime` request?

Reply:
Personal I don't like add this to `InitRuntime`, but I saw `skip_ra` already there...